R/ggplotly.R

```diff
@@ -217,8 +217,24 @@ gg2list <- function(p){
     }
     
     # This extracts essential info for this geom/layer.
-    traces <- layer2traces(L, df, misc)
-
+    if (L$geom$objname == "smooth") {
+      # smooth is really a line + ribbon geom
+      # draw ribbon (unless se = FALSE)
+      misc$smoothRibbon <- TRUE
+      trace1 <- if (isTRUE(L$stat_params$se == FALSE)) {
+        NULL
+      } else {
+        layer2traces(L, df, misc)
+      }
+      misc$smoothRibbon <- FALSE
+      # always draw the line
+      misc$smoothLine <- TRUE
+      trace2 <- layer2traces(L, df, misc)
+      trace2 <- lapply(trace2, function(x) { x$showlegend <- FALSE; x })
```

Comment:
@tdhock What?! I would say that, in R, you want to use *ply functions (in order to avoid for loops) as much as possible!!